src/main.rs

```diff
@@ -15,72 +15,73 @@ fn main() -> io::Result<()> {
+                            println!("nothing to do for now \n");
+                            io::stdout().flush()?;
+                        } else {
+                            for todo in lines {
+                                println!("{}", todo)
+                            }
                         }
                     }
+                    Err(why) => eprintln!("Error reading file : {}", why),
                 }
-                Err(why) => eprintln!("Error reading file : {}", why),
+                println!("\n");
             }
-            println!("\n");
-
-            continue;
-        }
-
-        if input.trim() == "2" {
-            let mut new_todo = String::new();
-            println!("please input new todo >> \n");
-            io::stdout().flush()?;
-            io::stdin().read_line(&mut new_todo)?;
+            "2" => {
+                let mut new_todo = String::new();
+                println!("please input new todo >> \n");
+                io::stdout().flush()?;
+                io::stdin().read_line(&mut new_todo)?;
 
-            match try_add_lines(&new_todo) {
-                Err(why) => eprintln!("couldn't add {} : {}", new_todo, why),
-                Ok(_) => println!("success!\n"),
+                match try_add_lines(&new_todo) {
+                    Err(why) => eprintln!("couldn't add {} : {}", new_todo, why),
+                    Ok(_) => println!("success!\n"),
+                }
             }
-            continue;
-        }
+            "3" => {
+                let to_dos = match try_read_lines("todo.txt") {
+                    Ok(lines) => lines,
+                    Err(e) => {
+                        eprintln!("Couldn't read file!! : {}", e);
+                        continue;
+                    }
+                };
 
-        if input.trim() == "3" {
-            if try_read_lines("todo.txt")?.len() == 0 {
-                println!("Nothing to delete now \n");
-                io::stdout().flush()?;
-            } else {
-                println!("which one? >");
+                if to_dos.len() == 0 {
+                    println!("Nothing to delete now \n");
+                    io::stdout().flush()?;
+                } else {
+                    println!("which one? >");
 
-                if let Ok(vector) = try_read_lines("todo.txt") {
-                    for todo in vector {
-                        println!("{}", todo)
+                    for todo in to_dos {
+                        println!("{}", todo);
                     }
-                } else {
-                    eprintln!("couldn't read file!!!")
-                }
+                };
 
                 let mut delete_todo = String::new();
 
                 io::stdin().read_line(&mut delete_todo)?;
 
                 match find_index_opt(&delete_todo) {
-                    Some(_) => {
+                    Ok(_) => {
                         try_update_file(&delete_todo)?;
```

Comment:
If you return `Optional`, I think you need to handle the case that the function returns None.
In this case, when the function returns None, this flow goes to print successfully~.

Reply:
You mean it is possible that the function returns `OK` but `None`, right? If so, I can change the code to either return `Result<usize, Error>` in `find_index_opt()` or handle what to do if return `None`,  I think the latter is better in point of detailed error handling. I would appreciate if you give me an advice.

Reply:
You should write the latter way.

Reply:
The error will occur when reading file process is failed. The None will be returned the input word is not found.

Reply:
You put `?` many points. Do you understand what it means? You might misunderstand the difference between try-catch and result or optional.
https://doc.rust-lang.org/rust-by-example/std/result/question_mark.html

Reply:
My understanding for `?`  is shorthand of `match` expression for concise code. try-catch is if an error happens in try block, jump into catch block and not execution in try block. Result is explicity implement what to do when error occurs, Optional is return value => `Some`, no value => `None`

Reply:
OK, you understand well. By the way, won't you handle `try_update_file`?

Reply:
You mean handling Error and Ok explicitliy, like below? If so, yes, I think I should do so.
From this,
```Rust
src/main.rs
match find_index_opt(&delete_todo) {
                    Ok(index) => match index {
                        None => println!("not found in the list"),

                        Some(_) => {
                            try_update_file(&delete_todo)?;
                            println!("successfully delete {} ", delete_todo);
                        }
                    },
                    Err(why) => eprintln!("couldn't read the file : {}", why),
                }

```

to this
```Rust
 match find_index_opt(&delete_todo) {
                    Ok(index) => match index {
                        None => println!("not found in the list"),

                        Some(_) => match try_update_file(&delete_todo) {
                            Err(why) => eprintln!("failed to update the file : {}", why),
                            Ok(_) => println!("successfully delete {}", delete_todo),
                        },
                    },
                    Err(why) => eprintln!("couldn't read the file : {}", why),
                }

```

Reply:
Thank you for confirming about thi verbally today, I already updated it.